### ISSUE TITLE
Generated Converter Type Inference

### ISSUE THREAD

**Issue by lachlanwgordon** [mfractor/mfractor-feedback#156]
If I have a binding and I write the name of a converter that doesn't exist, in the generate context menu I can select "Generate a new value converter that converts 'Type' into a 'TextAlignment'" but the converter it generates just has 'object' for the input and output types when casting.

e.g.
I declare a converter that doesn't exist in xaml, context menu exposes mfractor generating one for me.
![Screen Shot 2019-12-05 at 4 37 35 pm](https://user-images.githubusercontent.com/29908924/70207396-c7957380-177e-11ea-9988-929fb17825d3.png)

The generated code checks if `value is object` and casts the output to `object` but the action selected from the context menu implies it will know the types. The Attribute also has object for both values when you should be able to guess the real types.

```
    [ValueConversion(typeof(object), typeof(object))]
    public class TypeToTextAlignmentConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value is object == false)
            {
                return default(object);
            }

            var input = (object)value;

            // TODO: Put your value conversion logic here.

            return default(object);
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
```

The Wizard knows how to infer types from string names so hopefully you can use that code.

**Comment by matthewrdev** [comment]
@lachlanwgordon Thanks for reporting this issue. This code fix should ideally use the value converter wizard as it supports what you’re describing.

On a side note, do you have any feedback on the automatic type inference? Is it working naturally for you and makes sense?

**Comment by lachlanwgordon** [comment]
Works great if both types are primatives(or low level system classes).

If I'm using my own classes of something from Xamarin Forms it can't guess which is probably fair enough because there would be a lot of searching for you to do.

One quirk that's worth fixing is, if input is not inferrable but output is, it ignores my input type and puts the output type in the input type field.

e.g.

I named converter `TypeToStringConverter` it fails to infer that I mean System.Type(which is fine*) but it infers String is my input type, which should be the output type.
![Screen Shot 2019-12-05 at 5 33 16 pm](https://user-images.githubusercontent.com/29908924/70210066-e0555780-1785-11ea-95f2-a4a40b65c121.png)

*I reckon you probably can find System.Type given that you named a folder System.Type in bug #158, but in general, for this issue it's a fair assumption that the inference will often fail and you should be able to recover for the output.

**Comment by matthewrdev** [comment]
@lachlanwgordon This has been fixed and will be available in MFractor 3.9.1
